corehq/apps/api/resources/v0_5.py

```diff
@@ -1087,3 +1088,23 @@ def prepend_urls(self):
             url(r"^(?P<resource_name>{})/(?P<config_id>[\w\d_.-]+)/feed".format(
                 self._meta.resource_name), self.wrap_view('dispatch_list')),
         ]
+
+
+class MessagingEventResource(HqBaseResource, ModelResource):
+
+    class Meta(object):
+        queryset = MessagingEvent.objects.all()
+        list_allowed_methods = ['get']
+        detail_allowed_methods = ['get']
+        resource_name = 'messaging-event'
+        authentication = RequirePermissionAuthentication(Permissions.edit_data)
```

Comment:
any reason for the `edit_data` permission?

Reply:
Nope, just copied from [this guy](https://github.com/dimagi/commcare-hq/pull/3562/files#diff-efc09ad10beb9e39276a54e59070e11ec9947779bc6a4413196f0a086e70c841R350). It wasn't totally clear to me what it should be, and figured erring on the side of more-restrictive at first made sense.

Reply:
@snopoke Do you think I should change this to something else?

Reply:
Fine for now. In the long run I think anyone who can access the messaging reports should be able to access the API as well.